sklearn/decomposition/dict_learning.py

```diff
@@ -264,6 +299,15 @@ def sparse_encode(X, dictionary, gram=None, cov=None, algorithm='lasso_lars',
         regularization = n_nonzero_coefs
         if regularization is None:
             regularization = min(max(n_features / 10, 1), n_components)
+    elif algorithm == 'elastic_net':
+        if alpha is None and l1_ratio is None:
+            regularization = (1., 1.)
+        elif alpha is not None and l1_ratio is None:
+            regularization = (alpha, 1.)
+        elif alpha is None and l1_ratio is not None:
+            regularization = (1., l1_ratio)
+        else:
+            regularization = (alpha, l1_ratio)
```

Comment:
This can be tidied:
```python
reg_alpha= 1 if alpha is None else alpha
reg_l1_ratio = 1 if l1_ratio if None else l1_ratio
regularization = (reg_alpha, reg_l1_ratio)
```